pkg/clusteragent/admission/controllers/webhook/common.go

```diff
@@ -8,13 +8,59 @@
 package webhook
 
 import (
+	"encoding/json"
+	agentsidecar "github.com/DataDog/datadog-agent/pkg/clusteragent/admission/mutate/agent_sidecar"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
 	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/common"
 	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/mutate"
 	"github.com/DataDog/datadog-agent/pkg/config"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Selector specifies an object label selector and a namespace label selector
+type Selector struct {
+	ObjectSelector    metav1.LabelSelector `yaml:"objectSelector,omitempty"`
+	NamespaceSelector metav1.LabelSelector `yaml:"namespaceSelector,omitempty"`
+}
+
+// buildAgentSidecarObjectSelectors returns the mutating webhooks object selectors based on the configuration
+func buildAgentSidecarObjectSelectors() (namespaceSelector, objectSelector *metav1.LabelSelector) {
+	// Read and parse selectors
+	selectorsJSON := config.Datadog.GetString("admission_controller.agent_sidecar.selectors")
+
+	var selectors []Selector
+
+	err := json.Unmarshal([]byte(selectorsJSON), &selectors)
+	if err != nil {
+		log.Errorf("failed to parse selectors for admission controller agent sidecar injection webhook: %s", err)
+		return nil, nil
+	}
+
+	if len(selectors) > 0 {
```

Comment:
This is how I see it now, we haven't agreed yet how this will be extended some take this with some grain of salt.

When customers upgrade to the next version of sidecar injection we will have to make fewer assumptions about existing configs, i.e. we will always have one selector, one override at most. In that case we would handle this 'special case' by initializing new fields in the structs with some defaults e.g. `selector.name: default`, `profile.selectors: [default]` and new code would just handle it as a normal case. If we have multiple selectors or profiles - that wouldn't would. 
Forcing smaller input space (without limiting functionality) will make the change easier in the future.

Reply:
Makes sense

PR updated

Thanks